qtaskbarcontrol_win.cpp

```diff
@@ -3,16 +3,20 @@
 #include <QPainter>
 #include <QWinTaskbarProgress>
 
-QTaskbarControlPrivate *QTaskbarControlPrivate::createPrivate(QTaskbarControl *q_ptr)
+QTaskbarControlPrivate *QTaskbarControlPrivate::createPrivate()
 {
-	return new QWinTaskbarControl{q_ptr};
+	return new QWinTaskbarControl;
 }
 
-QWinTaskbarControl::QWinTaskbarControl(QTaskbarControl *q_ptr) :
-	_q_ptr{q_ptr},
-	_button{new QWinTaskbarButton{q_ptr}}
+QWinTaskbarControl::QWinTaskbarControl() :
+	_button{new QWinTaskbarButton}
 {}
 
+QWinTaskbarControl::~QWinTaskbarControl()
+{
+	delete _button;
+}
+
```

Comment:
I don't like the manual deletetion, just to not pass the `q_ptr` please undo that change and keep the `new QWinTaskbarButton{q_ptr}`. It's cleaner, as I try to eliminate any `delete` from my code wherever possible.

Reply:
What if I use `QScopedPointer` for `QWinTaskbarButton`? This will allow to avoid extra parameter for all other platforms, will avoid manual deletion and `_button` will be destroyed with pimpl object instead after `QTaskbarControl` destruction itself.

Reply:
Still, I prefer it that way, but if you insist, go ahead and use `QScopedPointer`

Reply:
No problem I will revert the commit :)